substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaNetNetUtil.java

```diff
@@ -223,6 +223,114 @@ static InetAddress NET_SockaddrToInetAddress(Socket.sockaddr him, CIntPointer po
+                    // 274 return JNI_FALSE;
+                    return false;
+                }
+                // 276 addrNew = NET_IPv4MappedToIPv4(caddrNew);
+                addrNew = JavaNetNetUtilMD.NET_IPv4MappedToIPv4(caddrNew);
+                // 277 addrCur = getInetAddress_addr(env, iaObj);
+                addrCur = JavaNetNetUtilMD.getInetAddress_addr(iaObj);
+                // 278 JNU_CHECK_EXCEPTION_RETURN(env, JNI_FALSE);
+                // 279 if (addrNew == addrCur) {
+                if (addrNew == addrCur) {
+                    // 280 return JNI_TRUE;
+                    return true;
+                } else {
+                    // 282 return JNI_FALSE;
+                    return false;
+                }
+                // 284 } else {
+            } else {
+                // 285 jbyteArray ipaddress;
+                byte[] ipaddr;
+                // 286 jbyte caddrCur[16];
+                CCharPointer caddrCur = StackValue.get(16);
+                // 287 int scope;
+                int scope;
+                // 289 if (family == AF_INET) {
+                if (family == Socket.AF_INET()) {
+                    // 290 return JNI_FALSE;
+                    return false;
+                }
+                // 292 scope = getInet6Address_scopeid(env, iaObj);
+                scope = getInet6Address_scopeid((Inet6Address) iaObj);
+                // 293 getInet6Address_ipaddress(env, iaObj, (char *)caddrCur);
+                getInet6Address_ipAddress((Inet6Address) iaObj, caddrCur);
+                // 294 if (NET_IsEqual(caddrNew, caddrCur) && cmpScopeID(scope, him)) {
+                if (JavaNetNetUtilMD.NET_IsEqual(caddrCur, caddrCur) && JavaNetNetUtilMD.cmpScopeID(scope, him)) {
+                    // 295 return JNI_TRUE;
+                    return true;
+                } else {
+                    // 297 return JNI_FALSE;
+                    return false;
+                }
+            }
+            // 301 #endif /* AF_INET6 */
+        }
+
+        // 303 struct sockaddr_in *him4 = (struct sockaddr_in *)him;
+        NetinetIn.sockaddr_in him4 = (NetinetIn.sockaddr_in) him;
+        // 304 int addrNew, addrCur;
+        int addrNew, addrCur;
+
+        // 305 if (family != AF_INET) {
+        if (family != Socket.AF_INET()) {
+            // 306 return JNI_FALSE;
+            return false;
+        }
+
+        // 308 addrNew = ntohl(him4->sin_addr.s_addr);
+        addrNew = NetinetIn.ntohl(him4.sin_addr().s_addr());
+        // 309 addrCur = getInetAddress_addr(env, iaObj);
+        addrCur = JavaNetNetUtilMD.getInetAddress_addr(iaObj);
+        // 310 JNU_CHECK_EXCEPTION_RETURN(env, JNI_FALSE);
+        // 311 if (addrNew == addrCur) {
+        if (addrNew == addrCur) {
+            // 312 return JNI_TRUE;
+            return true;
+        } else {
+            // 314 return JNI_FALSE;
+            return false;
+        }
+    }
+
     static boolean isIPv4Mapped(CCharPointer caddr) {
```

Comment:
Not your fault: This method should almost certainly move to `JavaNetNetUtilMD.NET_IsIPv4Mapped`.  I will file an issue to move it, either as a separate commit when I merge your changes, or later.